pkg/reconciler/clusterducktype/clusterducktype.go

```diff
@@ -34,15 +38,21 @@ import (
 // Reconciler implements ducktypereconciler.Interface for
 // ClusterDuckType resources.
 type Reconciler struct {
+	client    kubernetes.Interface
 	crdLister apiextensionslisters.CustomResourceDefinitionLister
+
+	resourceMapper collection.ResourceMapper
+	rmx            sync.Mutex
```

Comment:
I think if you use sync.RWMutex you might be better off. In the ReconcileKind you have just grab the RLock, and threads won't block each other. They will only wait when the Write lock is taken...

Reply:
and yes, I think you will need to copy the r.resourceMapper when creating the hunter

Reply:
Will do next.

Reply:
ok